File: ocaml/exemples/exemple_file_de_priorite.ml
let chaine_de_liste_entiere valeurs =
  "[" ^ String.concat " ; " (List.map string_of_int valeurs) ^ "]"

let () =
  let file_de_priorite =
    File_de_priorite.creer_file_de_priorite ()
    |> fun file_de_priorite -> File_de_priorite.inserer_element_file_de_priorite file_de_priorite 7
    |> fun file_de_priorite -> File_de_priorite.inserer_element_file_de_priorite file_de_priorite 2
    |> fun file_de_priorite -> File_de_priorite.inserer_element_file_de_priorite file_de_priorite 9
    |> fun file_de_priorite -> File_de_priorite.inserer_element_file_de_priorite file_de_priorite 4
  in
  Printf.printf "File de priorite initiale : %s\n"
    (chaine_de_liste_entiere file_de_priorite);
  match File_de_priorite.supprimer_element_prioritaire_file_de_priorite file_de_priorite with
  | None -> print_endline "Aucun element prioritaire."
  | Some (valeur, reste) ->
      Printf.printf "Element prioritaire retire : %d\n" valeur;
      Printf.printf "File restante : %s\n" (chaine_de_liste_entiere reste)

File: c/exemples/exemple_file_de_priorite.c
#include "file_de_priorite.h"

#include <stdio.h>

int main(void) {
    FileDePriorite file_de_priorite = creer_file_de_priorite(4U);
    int valeur;

    inserer_element_file_de_priorite(&file_de_priorite, 7);
    inserer_element_file_de_priorite(&file_de_priorite, 2);
    inserer_element_file_de_priorite(&file_de_priorite, 9);
    inserer_element_file_de_priorite(&file_de_priorite, 4);

    if (obtenir_element_prioritaire_file_de_priorite(&file_de_priorite, &valeur)) {
        printf("Element prioritaire initial : %d\n", valeur);
    }

    while (supprimer_element_prioritaire_file_de_priorite(&file_de_priorite, &valeur)) {
        printf("Element retire : %d\n", valeur);
    }

    detruire_file_de_priorite(&file_de_priorite);
    return 0;
}
